client/classic/upe/payment-processing.js

```diff
@@ -326,3 +333,61 @@ export const createAndConfirmSetupIntent = (
 			return confirmedSetupIntent;
 		} );
 };
+
+/**
+ * Handles the checkout for Voucher payment methods (Boleto & Oxxo).
+ *
+ * This function manually processes the checkout and displays the voucher
+ * to the customer, before then redirecting to the order received page.
+ *
+ * @param {Object} paymentMethod The payment method object.
+ * @param {Object} jQueryForm    The jQuery object for the form being submitted.
+ * @param {Object} api           The API object used to create the Stripe payment method.
+ */
+const handleVoucherCheckout = ( paymentMethod, jQueryForm, api ) => {
+	const formFields = jQueryForm.serializeArray().reduce( ( obj, field ) => {
+		obj[ field.name ] = field.value;
+		return obj;
+	}, {} );
+
+	/**
+	 * Manually process the checkout form rather than submitting the form so that we can display
+	 * the Boleto/Oxxo voucher on checkout page, before redirecting to the order received page.
+	 */
+	api.processCheckout( paymentMethod.id, formFields )
```

Comment:
Marking this PR as on-hold as I need to rethink this approach because it doesn't work for the Pay for Order page.

Reply:
I've just pushed up [30abdd0](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2823/commits/30abdd03e188c394330f495a794f5a9fac46af7e) which adds support for using Boleto and Oxxo on the Pay for Order page.

This unfortunately required a much bigger refactor of my original PR than I anticipated. I've tried to include as much inline comments as possible to help make sense of the approach.